src/commons/BrowserUtils.ts

```diff
@@ -234,7 +238,7 @@ export namespace BrowserUtils {
    * @param options WaitForOptions options (optional) { timeout, timeoutMsg, interval }
    */
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  export function waitUntil(condition: () => any, options?: WaitUntilOptions): any {
+  export async function waitUntil(condition: () => Promise<any>, options?: WaitUntilOptions): Promise<any> {
     return browser.waitUntil(() => condition(), { ...{ timeout: DEFAULT_TIME_OUT }, ...options });
```

Comment:
need to wait for condition, something like this
`    return browser.waitUntil(async () => await condition(), { ...{ timeout: DEFAULT_TIME_OUT }, ...options });
`

Reply:
changed condition type to boolean | Promise<boolean>